src/sentry/runner/commands/run.py

```diff
@@ -505,6 +505,17 @@ def occurrences_ingest_consumer(**options):
         run_processor_with_signals(consumer)
 
 
+def delay_kafka_rebalance(consumer_name: str) -> None:
+    """
+    get current time.
+    get the seconds part of the current time.
+
+    get the seconds delay to sleep from an option.
+
+    and then continue to sleep for 0.5 seconds until you get to the exact second when you want to start/stop the app.
```

Comment:
0.5 seconds or something granular just so that we don't accidentally miss the next rebalance